cli/exchange/user.go

```diff
@@ -76,6 +94,35 @@ func UserSetAdmin(org, userPwCreds, user string, isAdmin bool) {
 	cliutils.ExchangePutPost("Exchange", http.MethodPatch, cliutils.GetExchangeUrl(), "orgs/"+org+"/users/"+user, cliutils.OrgAndCreds(org, userPwCreds), []int{201}, patchUserReq, nil)
 }
 
+func UserSetHubAdmin(org, userPwCreds, user string, isHubAdmin bool) {
+	// get message printer
+	msgPrinter := i18n.GetMessagePrinter()
+
+	_, userId := cliutils.TrimOrg(org, user)
+
+	// Get the user from the exchange
+	var userList ExchangeUsers
+	httpCode := cliutils.ExchangeGet("Exchange", cliutils.GetExchangeUrl(), "orgs/root/users/"+userId, cliutils.OrgAndCreds(org, userPwCreds), []int{200, 404}, &userList)
+	if httpCode == 200 {
+		// If the user already exists in the root org, we need to update it with a PUT to set hubAdmin to correct state
+		exchUser := cliutils.AddOrg("root", userId)
+		if existingUser, ok := userList.Users[exchUser]; !ok {
+			cliutils.Fatal(cliutils.NOT_FOUND, msgPrinter.Sprintf("user '%s' not found in list returned from exchange", exchUser))
+		} else {
+			updatedUser := cliutils.UserExchangeReq{Password: existingUser.Password, Admin: existingUser.Admin, HubAdmin: isHubAdmin, Email: existingUser.Email}
+			cliutils.ExchangePutPost("Exchange", http.MethodPut, cliutils.GetExchangeUrl(), "orgs/root/users/"+userId, cliutils.OrgAndCreds(org, userPwCreds), []int{201}, updatedUser, nil)
+		}
+	} else if isHubAdmin {
+		// If the user does not exist in the root org and the user isHubAdmin, create them with hubAdmin set to true and no other fields
+		postUserReq := cliutils.UserExchangeReq{HubAdmin: true}
+		httpCode := cliutils.ExchangePutPost("Exchange", http.MethodPost, cliutils.GetExchangeUrl(), "orgs/root/users/"+userId, cliutils.OrgAndCreds(org, userPwCreds), []int{201, 400}, postUserReq, nil)
+		if httpCode == 400 {
+			cliutils.Fatal(cliutils.CLI_GENERAL_ERROR, msgPrinter.Sprintf("Only exchange users in the root org can be hub admins."))
```

Comment:
I dont understand this error message. We get here because of a 404 on the GET, but we received a 400 because why? we already forced the user into root org based on how the API is being called.

Reply:
The 400 is what we get if we try to call this post on a user who does not exist in the root org, and is not an IAM user. This post with the email and some other fields blank is only allowed for IAM users. This isn't really clear from the error message here, but I am trying to avoid saying "IAM" in the open source